### DISCUSSION TITLE
Light/Dark mode settings?

### DISCUSSION THREAD

**Discussion by cmacrae** [cmacrae/spacebar#67] [Ideas]
☀️ Would people be interested in light/dark mode settings? 🌑  
I achieved something like this in the past with a hacky launchd script which I [wrote about here](https://cmacr.ae/post/2020-05-03-keeping-up-appearances-non-native-macos-light-dark-mode/) (generally, I wouldn't recommend this approach...). It might be nice to have a native feature for this 🤔 

Maintaining double the `*_color` settings for such circumstances could prove cumbersome... I was thinking maybe it'd be easy to overcome this by invoking a separate config file/script when the mode changed, but then I wouldn't want this being used as a general purpose tool to invoke arbitrary actions upon mode change. With that, I'm considering writing a general purpose tool to do just that, so maybe such functionality could simply be delegated to that, as a separate project 🤷 

Feel free to share your thoughts!

**Comment by knowler** [comment]
I would be interested in this. Right now I have a command that switches between light and dark mode that just includes changing the foreground and background for spacebar ([code](https://github.com/knowler/dotfiles/blob/40eb8754ef206580acd38876211a8fff74c27799/.local/bin/lightswitch#L15-L18)).

**Reply by basbebe** [reply]
I'm doing something similar:
- use [DarkModeBuddy](https://github.com/insidegui/DarkModeBuddy) to change mode according to ambient light.
- use [dark-mode-notify](https://github.com/bouk/dark-mode-notify) to run [a script](https://github.com/basbebe/dotfiles/blob/master/.config/fish/functions/change_theme.fish) when mode changes
- [this script](https://github.com/basbebe/dotfiles/blob/master/.config/spacebar/change_mode.sh) then changes background and foreground of spacebar
